proof/file.go

```diff
@@ -222,6 +222,23 @@ func (f *File) ProofAt(index uint32) (*Proof, error) {
 	return proof, nil
 }
 
+// LocateProof calls the given predicate for each proof in the file and returns
+// the first proof (and the index) where the predicate returns true. If no proof
+// is found, this returns ErrNoProofAvailable.
+func (f *File) LocateProof(cb func(*Proof) bool) (*Proof, uint32, error) {
+	for i := 0; i < f.NumProofs(); i++ {
```

Comment:
Perhaps this should scan _backwards_ instead? Since most of the time we're only interested in the last few proofs when a re-org happens. Could add a bool here, or expose via another method.